pandas/core/generic.py

```diff
@@ -2427,22 +2427,26 @@ def copy(self, deep=True):
         data = self._data.copy(deep=deep)
         return self._constructor(data).__finalize__(self)
 
-    def convert_objects(self, convert_dates=True, convert_numeric=False,
-                        convert_timedeltas=True, copy=True):
+    @deprecate_kwarg(old_arg_name='convert_dates', new_arg_name='datetime')
+    @deprecate_kwarg(old_arg_name='convert_numeric', new_arg_name='numeric')
+    @deprecate_kwarg(old_arg_name='convert_timedeltas', new_arg_name='timedelta')
+    def convert_objects(self, datetime=True, numeric=False,
```

Comment:
hmm, I think to be consistent, let's make ``datetime=False`` by-default (you can simply specify it as ``True`` where needed for soft-conversions), but from a user perspective this would be confusing. Then I would raise if all three are ``False`` (IOW the user didn't specify any type of conversions)